src/admin_extra/service_admin_extra_migration.erl

```diff
@@ -0,0 +1,788 @@
+%%% ==================================================================
+%%% @doc
+%%% MongooseIM migration
+%%% @end
+%%% ==================================================================
+
+-module(service_admin_extra_migration).
+
+-export([commands/0, migrate/3, get_all_data/1]).
+
+-export([
+  migrate_pubsub_nodes/1,
+  migrate_pubsub_affiliations/1,
+  migrate_pubsub_subscriptions/1,
+  migrate_pubsub_items/1,
+  migrate_vcard/1,
+  migrate_vcard_search/1,
+  migrate_users/1,
+  migrate_event_pusher_push_subscription/1,
+  migrate_rosterusers/1,
+  migrate_roster_version/1,
+  migrate_rostergroups/1,
+  migrate_last/1,
+  migrate_private_storage/1,
+  migrate_offline_message/1,
+  migrate_muc_light_rooms/1
+]).
+
+%%% ==================================================================
+%%% Records
+%%% ==================================================================
+
+-record(passwd, {us, password}).
+
+-record(push_subscription, {
+  user_jid    :: jid:simple_bare_jid() | undefined,
+  pubsub_jid  :: jid:jid(),
+  pubsub_node :: binary(),
+  form        :: [{Name :: binary(), Value :: binary()}]
+}).
+
+-record(private_storage, {usns, xml}).
+
+-record(muc_light_room, {
+  room :: jid:simple_bare_jid(),
+  config :: [{atom(), term()}],
+  aff_users :: [{{LUser :: binary(), Lserver :: binary()}, owner | member | none}],
+  version :: binary()
+}).
+
+%%% ==================================================================
+%%% Macros
+%%% ==================================================================
+
+-define(HOST, (hd(ejabberd_config:get_global_option(hosts)))).
```

Comment:
This is already defined in: https://github.com/esl/MongooseIM/blob/6b0903f830f7f100b03461c6868d8a7ed7445546/include/mongoose.hrl#L26